universe/ValueRefs.cpp

```diff
@@ -234,50 +235,45 @@ namespace {
         mutable UniverseObjectType m_type;
     };
 
-    const std::map<std::string, MeterType>& GetMeterNameMap() {
-        static const std::map<std::string, MeterType> meter_name_map{
-            {"Population",           MeterType::METER_POPULATION},
-            {"TargetPopulation",     MeterType::METER_TARGET_POPULATION},
-            {"Industry",             MeterType::METER_INDUSTRY},
-            {"TargetIndustry",       MeterType::METER_TARGET_INDUSTRY},
-            {"Research",             MeterType::METER_RESEARCH},
-            {"TargetResearch",       MeterType::METER_TARGET_RESEARCH},
-            {"Influence",            MeterType::METER_INFLUENCE},
-            {"TargetInfluence",      MeterType::METER_TARGET_INFLUENCE},
-            {"Construction",         MeterType::METER_CONSTRUCTION},
-            {"TargetConstruction",   MeterType::METER_TARGET_CONSTRUCTION},
-            {"Happiness",            MeterType::METER_HAPPINESS},
-            {"TargetHappiness",      MeterType::METER_TARGET_HAPPINESS},
-            {"MaxFuel",              MeterType::METER_MAX_FUEL},
-            {"Fuel",                 MeterType::METER_FUEL},
-            {"MaxStructure",         MeterType::METER_MAX_STRUCTURE},
-            {"Structure",            MeterType::METER_STRUCTURE},
-            {"MaxShield",            MeterType::METER_MAX_SHIELD},
-            {"Shield",               MeterType::METER_SHIELD},
-            {"MaxDefense",           MeterType::METER_MAX_DEFENSE},
-            {"Defense",              MeterType::METER_DEFENSE},
-            {"MaxTroops",            MeterType::METER_MAX_TROOPS},
-            {"Troops",               MeterType::METER_TROOPS},
-            {"RebelTroops",          MeterType::METER_REBEL_TROOPS},
-            {"Supply",               MeterType::METER_SUPPLY},
-            {"MaxSupply",            MeterType::METER_MAX_SUPPLY},
-            {"Stockpile",            MeterType::METER_STOCKPILE},
-            {"MaxStockpile",         MeterType::METER_MAX_STOCKPILE},
-            {"Stealth",              MeterType::METER_STEALTH},
-            {"Detection",            MeterType::METER_DETECTION},
-            {"Speed",                MeterType::METER_SPEED},
-            {"Damage",               MeterType::METER_CAPACITY},
-            {"Capacity",             MeterType::METER_CAPACITY},
-            {"MaxCapacity",          MeterType::METER_MAX_CAPACITY},
-            {"SecondaryStat",        MeterType::METER_SECONDARY_STAT},
-            {"MaxSecondaryStat",     MeterType::METER_MAX_SECONDARY_STAT},
-            {"Size",                 MeterType::METER_SIZE}
-        };
-        return meter_name_map;
-    }
-    // force early init to avoid threading issues later
-    std::map<std::string, MeterType> dummy = GetMeterNameMap();
-
+    const std::array<std::pair<std::string, MeterType>, 36> NAME_TO_METER = {{
```

Comment:
I tried using a `constexpr` array with `std::string_view`, however the issue is `MeterToName` and the boost::phoenix wrapper which fails during linking if the return type is `std::string_view`.

Reply:
What is boost::phoenix issue? Maybe it could be solved with new Python FOCS parser?

Reply:
probably https://github.com/freeorion/freeorion/blob/master/parse/DoubleComplexValueRefParser.cpp#L10

Reply:
```cpp
    std::string MeterToNameWrapper(MeterType meter) {
        return std::string().append(ValueRef::MeterToName(meter));
    }

    BOOST_PHOENIX_ADAPT_FUNCTION(std::string, MeterToName_, MeterToNameWrapper, 1)
```

I defined the above wrapper. I'm not familiar with `BOOST_PHOENIX_ADAPT_FUNCTION`, but would a lambda work as well or does it have to be a named function?